core/libp2p/crypto/impl/crypto_provider_impl.cpp

```diff
@@ -0,0 +1,194 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "libp2p/crypto/impl/crypto_provider_impl.hpp"
+
+#include "libp2p/crypto/private_key.hpp"
+
+#include "libp2p/crypto/proto/keys.pb.h"
+
+#include <optional>
+
+namespace libp2p::crypto {
+  using kagome::common::Buffer;
+
+  Buffer CryptoProviderImpl::aesEncrypt(const common::Aes128Secret &secret,
+                                        const Buffer &data) const {
+    // TODO(yuraz):  implement
+    std::terminate();
+  }
+
+  Buffer CryptoProviderImpl::aesEncrypt(const common::Aes256Secret &secret,
+                                        const Buffer &data) const {
+    // TODO(yuraz):  implement
+    std::terminate();
+  }
+
+  Buffer CryptoProviderImpl::aesDecrypt(const common::Aes128Secret &secret,
+                                        const Buffer &data) const {
+    // TODO(yuraz):  implement
+    std::terminate();
+  }
+
+  Buffer CryptoProviderImpl::aesDecrypt(const common::Aes256Secret &secret,
+                                        const Buffer &data) const {
+    // TODO(yuraz):  implement
+    std::terminate();
+  }
+
+  Buffer CryptoProviderImpl::hmacDigest(common::HashType hash,
+                                        const Buffer &secret,
+                                        const Buffer &data) {
+    // TODO(yuraz):  implement
+    std::terminate();
+  }
+
+  common::KeyPair CryptoProviderImpl::generateEd25519Keypair() const {
+    // TODO(yuraz):  implement
+    std::terminate();
+  }
+
+  common::KeyPair CryptoProviderImpl::generateRSAKeypair(
+      common::RSAKeyType key_type) const {
+    // TODO(yuraz):  implement
+    std::terminate();
+  }
+
+  common::EphemeralKeyPair CryptoProviderImpl::generateEphemeralKeyPair(
+      common::CurveType curve) const {
+    // TODO(yuraz):  implement
+    std::terminate();
+  }
+
+  std::vector<common::StretchedKey> CryptoProviderImpl::keyStretcher(
+      common::CipherType cipher_type, common::HashType hash_type,
+      const Buffer &secret) const {
+    // TODO(yuraz):  implement
+    std::terminate();
+  }
+
+  namespace {
+    /**
+     * @brief converts common::KeyType to proto::KeyType
+     * @param key_type common key type value
+     * @return proto key type value
+     */
+    proto::KeyType marshalKeyType(common::KeyType key_type) {
+      switch (key_type) {
+        case common::KeyType::kUnspecified:
+          return proto::KeyType::kUnspecified;
+        case common::KeyType::kRSA1024:
+          return proto::KeyType::kRSA1024;
+        case common::KeyType::kRSA2048:
+          return proto::KeyType::kRSA2048;
+        case common::KeyType::kRSA4096:
+          return proto::KeyType::kRSA4096;
+        case common::KeyType::kED25519:
+          return proto::KeyType::kED25519;
+        default:
+          break;
+      }
+      BOOST_UNREACHABLE_RETURN(proto::KeyType)
```

Comment:
what does this mean?

Reply:
refactored unmarshal functions, no macros now.